lessons/12-redux/README.md

```diff
@@ -1 +1,83 @@
-#Lesson 12 - Redux
\ No newline at end of file
+## Table of Contents
+
+<!-- MarkdownTOC -->
+
+- [Lesson 12 - Redux](#lesson-12---redux)
+  - [History](#history)
+  - [Overview](#overview)
+  - [Integrate into React](#integrate-into-react)
+  - [Dev Tools](#dev-tools)
+
+<!-- /MarkdownTOC -->
+
+
+#Lesson 12 - Redux
+![Redux Logo](https://raw.githubusercontent.com/reactjs/redux/master/logo/logo-title-dark.png)
+[Redux](http://redux.js.org/) is a state management tool meant to be used on top of React.
+## History
+While Facebook was making React, they also came out with [Flux](https://github.com/facebook/flux).  Flux is a pattern for managing state and data flow in React applications.  It is not a strict framework, but more of an idea for a pattern that Facebook think developers should use.
+
+Since Flux is not its own framework, many different frameworks have come out to try and implement the Flux pattern.  Redux is not a strict implementation of Flux ([depending on who you ask](http://redux.js.org/docs/introduction/PriorArt.html#flux)), but it is heavily inspired by the Flux architecture.  We'll be using Redux for our apps in this class.  Unless you're curious, there's no need to learn Flux itself.
+
+## Overview
+### Three Core Principles
+1) The whole state of your app is contained in a single state object in a **store**.
+2) When the user does something that should change the state, an **action** is dispatched.
+3) The action is sent through a **reducer** function, which specifies how to change the state.
+
+### Todo Example
+If you're making a Todo app, the state of your app might look like this:
+```javascript
+{
+  filter: 'COMPLETED',
+  todos: [
+    {
+      text: 'brush teeth',
+      completed: true,
+    },
+    {
+      text: 'do my JS homework',
+      completed: false
+    }
+  ]
+}
+```
+We have two todos, one completed and one not, and the filter is set so that the user only sees the "completed" todos.
+
+Say the user changes the filter by pressing the "active" button.  The React/Redux app then dispatches an **action** that looks like this:
+```javascript
+var actionObj = {
```

Comment:
I might make you action examples follow [FSA](https://medium.com/@jtbennett/standard-actions-in-redux-c6a415c8aea4#.9m35maq2x) (flux standard action) format. I wouldn't mention the standard cause it might be a bit confusing. A few libraries expect actions to look that way so might as well learn it from the start.

the only difference would be
`var actionObj = {
type: 'CHANGE_FILTER,
payload: 'ACTIVE'
};`

or 

`var actionObj = {
type: 'CHANGE_FILTER,
payload: { filter: 'ACTIVE' }
};`